### DISCUSSION TITLE
Mongo adapter feasibility

### DISCUSSION THREAD

**Discussion by thinkingerrol** [tconbeer/harlequin#402] [Ideas (Feature Requests)]
I'm old school and feel comfortable with SQL but kind of shudder a little every time I have to query a NoSQL DB such as the MongoDB.
Just found out that MongoDB can in fact be queried with SQL:
https://studio3t.com/knowledge-base/articles/sql-query/

Do you guys think that a mongo adapter for harlequin might be technically feasible, and would you agree that it makes sense?

edit: In any case, Harlequin is *awesome*, I've heard about it on HN some days ago and I'm very successfully using it already with SQLite and PostgreSQL. It was super easy to install and get going, the F1 help is awesome and keyboard navigation is super intuitive so I got immediately very productive with it.

**Comment by tconbeer** [comment]
Never used Mongo. However, our requirements are:
1. A Python library (ideally DBAPI 2.0 ish) that can connect to the db and execute sql statements
2. A result set in one of several supported formats:

- A PyArrow Table or RecordBatch.
- A Mapping of str to Sequence, where keys represent column names and the sequences are the data in each column. For example: {"col_a": [1, 2, 3], "col_b": ["a", "b", "c"]}
- A Sequence of Iterables, like a list of tuples, representing rows of data (or records). Such a sequence MUST NOT contain a header row.
- A pathlib.Path or str path to a local Parquet file.


Right now there is additional limitation that the data table can only display rows of height 1, meaning that multiline json docs would not be displayed well in the table.
